core/lib/dal/src/storage_logs_dal.rs

```diff
@@ -502,6 +545,65 @@ impl StorageLogsDal<'_, '_> {
             .collect()
     }
 
+    // Retrieves all storage log entries for testing purposes.
+    pub async fn get_all_storage_logs_for_tests(&mut self) -> Vec<StorageLogDbRow> {
```

Comment:
I agree that "dump_(...)" is a nice convention, but I'm not yet a very big fan of the separate impl block. I suspect that as soon as such block gets bigger, it would be very easy to miss that you are adding something to test-only impl instead of one you intend to if you just add the method at the end of the impl